clippy_lints/src/matches.rs

```diff
@@ -361,7 +361,7 @@ fn all_ranges(cx: &LateContext, arms: &[Arm]) -> Vec<SpannedRange<ConstVal>> {
                 }
                 .filter_map(|pat| {
                     if_let_chain! {[
-                    let PatKind::Range(ref lhs, ref rhs) = pat.node,
+                    let PatKind::Range(ref lhs, ref rhs, _) = pat.node,
```

Comment:
I don't think that's quite right. The `SpannedRange` returned below is different depending on the inclusiveness